### DISCUSSION TITLE
[Question] RF module details

### DISCUSSION THREAD

**Discussion by qba667** [QuadCorei8085/elero_protocol#4] [Q&A] [answered]
@QuadCorei8085 great project - can you please let me know what radio module have you used?
My guess would be CC1101.

**Comment by QuadCorei8085** [comment, marked as answer]
cc1101 for testing but actually in the remote it is cc1100  the difference is subtle if you compare the datasheet

**Comment by qba667** [comment]
@QuadCorei8085 thanks a lot, probably after my Elero curtains will arrive I will spend some time on development of nice controller.

**Comment by QuadCorei8085** [comment]
I recently integrated the code to my thermostat project so far this code worked for me well.
I have 3 VarioTel2 remotes for 2+4+1 blinds. Used an esp32 and a cc1101.

Let me know how if you could used this or not, i'm curious to find out if other system is similar to mine.

**Comment by qba667** [comment]
@QuadCorei8085 Thanks, I will let you know as soon as I will get curtains delivered.

**Comment by qba667** [comment]
@QuadCorei8085 my setup should be mounted next week - it will be also VarioTel2 15 Channels remote with RolTop M-868.
For R&D purposed I bought the MonoTec-868 (still waiting for delivery) with single RolTop M-868.

**Reply by QuadCorei8085** [reply]
Nice, I have the 3x 5ch version of the VarioTel2 but I believe there won't be any difference between the 15ch and the RolTop.
One (among many) things I haven't checked is the "pairing" I think the motor learns the ID of the remote (maybe it can store multiple IDs? if you can use both monotec and variotel2 i guess this would be the case - the address is written to the flash of the remotes).

**Reply by qba667** [reply]
I think as well, it shall be same. Binding topic is important, if that is based on remote ID this seems to be huge security risk, trying all possible ID may control someone else curtains. I will try to dump this communication so maybe we can figure it out.
Interesting fact is also that the motor control unit contains 2 MCU - I need look closer at is, maybe they can be read out using this method:
https://www.rapid7.com/blog/post/2019/04/16/extracting-firmware-from-microcontrollers-onboard-flash-memory-part-1-atmel-microcontrollers/

**Reply by qba667** [reply]
@QuadCorei8085 other interesting functionality is time based control of curtains, I am wondering if remote is sending commands only or maybe configuration is store in motor control unit. If you have any other ideas just post them, I will try to figure it out.

**Reply by QuadCorei8085** [reply]
The encryption is purely based on the packet counter/packet ID and the payload. The address and other fields are completely irrelevant. So yes once you have the "encryption" (i believe "rolling code" is more fitting) algorithm you can control any blinds you only need to know is the addresses (and guess the button position in the payload, but if you record the address you can also just read out that one too).
In my case the complete neighbourhood with 100+ flats are equipped with such blinds. So if I had bad intentions the only thing i would need to do is record messages and filter for those where SRC==DST==HOP address (that is a remote) and replay them. In Elero's defense a shutter/blind/sunshade is not a security risk to the house/flat, it is just bad for the business if your remotes can be copied easily.

I think within the protocol there is no scheduling possible as neither the remote nor the blind has access to timebase. Elero offers this (elero centero?) gateway that has internet access and you could program your schedules over an app/web interface, but I read that they charge you monthly for the service (this was the reason i decided to hack the system, monthly subscription is imo the cancer of our society).

The above link only works in case the microcontrollers are not "locked".  The mcu in VarioTel2 is not locked so I would expect it is also not locked in the motors. But that is a lot of work to disassemble a C compiler generated binary and understand the behavior.

**Reply by qba667** [reply]
Thanks for next portion of details, there are remotes like elero TEMPOTEL 2 with build in scheduling - but they are limited to 10 channels.
Probably such remote has RTC and it sends commands at specific time - that would be my guess. I have also seen the gateway but I prefer to have open source solution - because of that I looked at same topic you did.  
Well I have some experience in de-assembling but not for AVR and as you discovered basic of the protocol probably we don't need it at all. 
Thanks for all details!

**Reply by andyboeh** [reply]
We have TempoTel 2 remotes, they have built-in RTC and send the radio signals at the specified times as you have guessed.
